shared/api/event.go

```diff
@@ -28,13 +29,60 @@ type Event struct {
 	Location string `yaml:"location,omitempty" json:"location,omitempty"`
 }
 
+// EventLogRecord represents single log record
+type EventLogRecord struct {
+	Time time.Time
+	Lvl  string
+	Msg  string
+	Ctx  []interface{}
+}
+
+// EventLogger is used for event logging purpose
+type EventLogger interface {
+	LogRecord(Event) (EventLogRecord, error)
+}
+
+// NewEventLogger creates specific instance of EventLogger interface
+func NewEventLogger(event Event) EventLogger {
+	var logger EventLogger
+	if event.Type == "logging" {
+		logger = &EventLogging{}
+	} else if event.Type == "lifecycle" {
+		logger = &EventLifecycle{}
+	}
+	return logger
+}
+
 // EventLogging represents a logging type event entry (admin only)
 type EventLogging struct {
 	Message string            `yaml:"message" json:"message"`
 	Level   string            `yaml:"level" json:"level"`
 	Context map[string]string `yaml:"context" json:"context"`
 }
 
+// LogRecord creates log record for the event of type `logging`
+func (e EventLogging) LogRecord(event Event) (EventLogRecord, error) {
```

Comment:
Can you drop the need to pass `event` to this function (and other one below) by recording `event` as part of the `EventLogging` struct (as a non-exported private field)?